zkevm-circuits/src/tx_circuit.rs

```diff
@@ -139,6 +445,559 @@ impl<F: Field> TxCircuitConfig<F> {
+            ]
+            .into_iter()
+            .zip(rlp_table.table_exprs(meta).into_iter())
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+
+        // lookup tx rlc(value).
+        meta.lookup_any("tx rlc(value) in RLPTable::TxSign", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                tag.value_equals(TxFieldTag::Value, Rotation::cur())(meta),
+            ]);
+            vec![
+                meta.query_advice(tx_table.tx_id, Rotation::cur()),
+                RlpTxTag::Value.expr(),
+                1.expr(), // tag_index == 1
+                meta.query_advice(tx_table.value, Rotation::cur()),
+                RlpDataType::TxSign.expr(),
+            ]
+            .into_iter()
+            .zip(rlp_table.table_exprs(meta).into_iter())
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+        meta.lookup_any("tx rlc(value) in RLPTable::TxHash", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                tag.value_equals(TxFieldTag::Value, Rotation::cur())(meta),
+            ]);
+            vec![
+                meta.query_advice(tx_table.tx_id, Rotation::cur()),
+                RlpTxTag::Value.expr(),
+                1.expr(), // tag_index == 1
+                meta.query_advice(tx_table.value, Rotation::cur()),
+                RlpDataType::TxHash.expr(),
+            ]
+            .into_iter()
+            .zip(rlp_table.table_exprs(meta).into_iter())
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+
+        // lookup to check CallDataLength of the tx's call data.
+        meta.lookup_any("tx calldatalength in TxTable", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                meta.query_advice(is_usable, Rotation::cur()),
+                tag.value_equals(TxFieldTag::CallData, Rotation::cur())(meta),
+                meta.query_advice(is_final, Rotation::cur()),
+            ]);
+            vec![
+                meta.query_advice(tx_table.tx_id, Rotation::cur()),
+                TxFieldTag::CallDataLength.expr(),
+                0.expr(),
+                meta.query_advice(tx_table.index, Rotation::cur()) + 1.expr(),
+            ]
+            .into_iter()
+            .zip(tx_table.table_exprs(meta).into_iter())
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+
+        // lookup to check CallDataGasCost of the tx's call data.
+        meta.lookup_any("tx calldatagascost in TxTable", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                meta.query_advice(is_usable, Rotation::cur()),
```

Comment:
It does not make sense to use `is_usable` to be part of the selector as `is_usable` is an advice column and , which means the malicious prover can set it to `zero` to disable relevant lookups.